core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/EraSubmissionPreparation.java

```diff
@@ -0,0 +1,496 @@
+      TransformerFactory.newInstance().newTransformer().transform(new DOMSource(document), new StreamResult(output));
+      output.closeEntry();
+    }
+
+    @Override
+    public boolean isEmpty() {
+      return items().count() > 0;
+    }
+
+    protected abstract Stream<T> items();
+
+    protected abstract void populate(Element itemElement, T item);
+
+    protected abstract String xmlCollectionName();
+
+    protected abstract String xmlName();
+
+  }
+
+  private static final DateFormat DF_TIMESTAMP = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+
+  private final String centreName;
+
+  private final Submission submission;
+
+  private final ChildSubmissionFile[] FILES = new ChildSubmissionFile[] { new StudyXmlSubfile(), new SampleXmlSubfile(),
+      new DilutionXmlSubfile(),
+      new ExperimentXmlSubfile() };
+
+  private final SubmissionActionType submissionAction;
+
+  private final User user;
+
+  public EraSubmissionPreparation(Submission submission, User user, String centreName, SubmissionActionType submissionAction) {
+    super();
+    this.submission = submission;
+    this.user = user;
+    this.centreName = centreName;
+    this.submissionAction = submissionAction;
+  }
+
+  public byte[] toBytes() {
+    try (ByteArrayOutputStream outputBytes = new ByteArrayOutputStream()) {
+
+      ZipOutputStream output = new ZipOutputStream(outputBytes);
+      submission.setSubmissionDate(new Date());
+
+      Document submissionDocument = DocumentBuilderFactory.newInstance().newDocumentBuilder().newDocument();
+
+      Element s = submissionDocument.createElementNS(null, "SUBMISSION");
+      s.setAttribute("alias", submission.getAlias());
+      s.setAttribute("submission_date", DF_TIMESTAMP.format(submission.getSubmissionDate()));
+      s.setAttribute("submission_comment", submission.getDescription());
+      s.setAttribute("center_name", centreName);
+
+      Element title = submissionDocument.createElementNS(null, "TITLE");
+      title.setTextContent(submission.getTitle());
+      s.appendChild(title);
+
+      Element contacts = submissionDocument.createElementNS(null, "CONTACTS");
+      Stream
+          .concat(Stream.of(user),
+              submission.getExperiments().stream().map(experiment -> experiment.getSecurityProfile().getOwner()).filter(Objects::nonNull))
+          .map(User::getFullName).distinct()
+          .map(contactName -> {
+            Element contact = submissionDocument.createElementNS(null, "CONTACT");
+            contact.setAttribute("name", contactName);
+            return contact;
+          }).forEach(contacts::appendChild);
+
+      s.appendChild(contacts);
+
+      Element actions = submissionDocument.createElementNS(null, "ACTIONS");
+
+      for (ChildSubmissionFile subFile : FILES) {
+
+        subFile.generateDocument(output);
+        if (!subFile.isEmpty()) {
+          if (submissionAction == SubmissionActionType.ADD || submissionAction == SubmissionActionType.VALIDATE) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)